### PR TITLE
Error if Flash tech is unusable

### PR DESCRIPTION
**Pull request by forbesjo**

### CODE REVIEW

File: src/plugin.js [hunk 1]
```diff
@@ -61,6 +61,13 @@ const initPlugin = function(player, options) {
   const resetMonitor = function() {
     window.clearTimeout(monitor);
     monitor = window.setTimeout(function() {
+      // if using Flash then make sure its API is available
+      if (/swf/i.test(player.tech_.el().data) &&
```
Comment (dmlap) [inline src/plugin.js]: Can we get at this without using `tech_`? That's supposed to be a private property.

Comment (forbesjo) [inline src/plugin.js]: Going to use `player.$()` instead

File: src/plugin.js [hunk 2]
```diff
@@ -61,9 +61,24 @@ const initPlugin = function(player, options) {
...
-        // if the player is paused or ended.
+      // never overwrite existing errors
+      if (player.error()) {
+        return;
+      }
+
+      let tech = player.$('.vjs-tech');
+
+      // don't error if using Flash and its API is available
+      if (tech && tech.type === 'application/x-shockwave-flash' && tech.vjs_getProperty) {
```
Comment (gkatsev) [inline src/plugin.js]: shouldn't this be `&& !tech.vjs_getProperty`?

Comment (forbesjo) [inline src/plugin.js]: you are correct

**Comment by forbesjo** [comment]
I think something's still wrong with this, here's the scenario:
Player is triggering timeupdates
Paused is true for some reason
Flash player is broken and its API is unavailable
-> Timeout will trigger in healthcheck, `resetMonitor` is not called
-> healthcheck triggers again but in `check` `resetMonitor` is called
  -> this will clear the existing timeout

File: src/plugin.js [hunk 3]
```diff
@@ -93,15 +93,31 @@ const initPlugin = function(player, options) {
...
+            !tech.vjs_getProperty) {
+          player.error({
+            code: -2,
+            type: 'PLAYER_ERR_TIMEOUT'
+          });
+          return;
+        }
+
+        // playback isn't expected if the player is paused, shut
+        // down monitoring
```
Comment (dmlap) [inline src/plugin.js]: I think this comment is somewhat inaccurate. Technically, you're saying "keep checking, it looks like everything is fine right now".

**Pull request merged**